[14.0] add service to retrieve elasticsearch settings

lmignon opened the pull request:
requires:

* [x] https://github.com/OCA/rest-framework/pull/205
* [x] https://github.com/OCA/rest-framework/pull/207

lmignon committed:
[IMP] shopinvader_elasticsearch: Export ES settings

lmignon committed:
[IMP] shopinvader: Allows access by authenticated users to public services methods

From base_rest>=14.0.4.0.2 a new predefined auth method can be declare in restapi.method declaration: 'public_or_default'. If used, when the HTTP route is generated, the auth method will be computed from the '_default_auth' property defined on the controller with 'public_or_' as prefix is such a method is available on the odoo instance or fallback on 'public'. The purpose of such a 'public_or_...' auth method is to provide a way to specify that a method should work for anonymous users but can be enhanced when an authenticated user is know.

sebastienbeau reviewed (approved):
LGTM (code review)

PierrickBrun reviewed (approved):
LGTM (code review)

simahawk commented on shopinvader_elasticsearch/services/settings.py:
```diff
@@ -0,0 +1,72 @@
+            }
+        }
+
+    @restapi.method(
```
This should probably be generic and defined into `shopinvader_search_engine` as `/search_engine` and `def search_engine_settings`. No?

lmignon replied:
@simahawk I prefer a specialized method by kind of search engine to be able to define a specific definition of the response according to the requirements of the specific search engine. This PR is still WIP since I'm working on base_rest to finalize the support of array response with Cerberus. With this last change we will get rid of the array encapsulation under a root key for all the specific settings methods...

lmignon committed:
[IMP] shopinvader: Rework 'settings' services since it's now possible to return array validated by cerberus  as body response

requires https://github.com/OCA/rest-framework/pull/207
requires https://github.com/OCA/rest-framework/pull/205

lmignon commented:
@sebastienbeau @PierrickBrun @simahawk I improved base_rest to be able to define better API by managing array as request/response body validated by cerberus and allowing documentation of properties into our cerberus schema . The last commit uses these new features.

lmignon commented:
/ocabot merge major
